### PR TITLE
Make HinSAGE reproducible

### PR DESCRIPTION
**Pull request by kjun9**
This is for both NAI and link prediction, which are the two tasks currently available for HinSAGE

Part of #749

### CODE REVIEW

**Comment by kjun9** [comment]
Hmm.. something is still flaky, investigating the test failure now

File: tests/mapper/test_node_mappers.py [hunk 1]
```diff
@@ -112,6 +112,44 @@ def example_hin_3(feature_size_by_type=None):
...
+    """
+
+    def flatten_features(seq):
+        # check (features == labels) and return flattened features
+        batches = [
+            (np.ravel(seq[i][0][0]), np.array(seq[i][1])) for i in range(len(seq))
+        ]
+        features, labels = zip(*batches)
+        features, labels = np.concatenate(features), np.concatenate(labels)
+        assert all(features == labels)
```
Comment (kieranricardo) [inline tests/mapper/test_node_mappers.py]: I think this can be:

```suggestion
        assert np.array_equal(features, labels)
```

File: tests/mapper/test_node_mappers.py [hunk 2]
```diff
@@ -112,6 +112,44 @@ def example_hin_3(feature_size_by_type=None):
...
+        features = flatten_features(seq)
+        seq.on_epoch_end()
+        features_next = flatten_features(seq)
+        return features, features_next
+
+    comparison_results = set()
+
+    for i in range(max_iter):
+        f1, f2 = consecutive_epochs(seq)
+        comparison_results.add(all(f1 == f2))
```
Comment (kieranricardo) [inline tests/mapper/test_node_mappers.py]: Same here:

```suggestion
        comparison_results.add(np.array_equal(f1, f2))
```

Comment (kieranricardo) [inline tests/mapper/test_node_mappers.py]: could the assertions happen in this line?

```suggestion
        if shuffle:
            assert not np.array_equal(f1, f2)
        else:
            assert np.array_equal(f1, f2)   
```

**Comment by Avalee21** [comment]
It should be blamed on random state use multiple times, which will generate different results. The following steps guarantee consistent results even at every epoch.
1. set stellargraph seed:  (1) `HinSAGELinkGenerator` (fix the seed in the `run` function of `SampledHeterogeneousBreadthFirstWalk` ) and (2) `flow` function 
2. [set tensorflow seed](https://stackoverflow.com/questions/50659482/why-cant-i-get-reproducible-results-in-keras-even-though-i-set-the-random-seeds/52897289#52897289)
